### PR TITLE
[iOS] Fix final state of canceled animations

### PR DESCRIPTION
**Pull request by dr1rrb**
fixes https://github.com/unoplatform/uno/issues/1956

## Bugfix
On iOS, if you change the visual state before the entrance animation of the current visual state is completed, and if the new state does not alter the animated property, then the property will be set to the final value of the animation instead of being rollback to the original value.

## What is the current behavior?
When canceled the `DoubleAnimation` pushes to its target property the `To` value.

## What is the new behavior?
When canceled the `DoubleAnnimation` restore the original value (note: not the `From` !), like WinUI.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other: Tooling
We can now assign a UI test to more than one category

### CODE REVIEW

File: doc/articles/uno-development/working-with-the-samples-apps.md [hunk 1]
```diff
@@ -3,11 +3,14 @@
 The Uno solution provides a set of sample applications that provide a way to test features, as
 well as provide a way to write UI Tests.
 
-Those applications are structured in a way that samples can created out of normal `UserControl` instances, marked with the `SampleControlInfoAttribute` so the sample application can discover them.
+Those applications are structured in a way that samples can created out of normal `UserControl` instances, 
+marked with the `SampleAttribute` (and previous `SampleControlInfoAttribute`) so the sample application can discover them.
```
Comment (carldebilly) [inline doc/articles/uno-development/working-with-the-samples-apps.md]: https://sembr.org/

File: src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml [hunk 2]
```diff
@@ -113,7 +113,7 @@
 			<DataTemplate x:Key="SampleTemplate">
 				<Border Style="{StaticResource TemplateContainerStyle}">
 					<u:StarStackPanel Orientation="Horizontal"
-									  Sizes="*,Auto">
+									  Sizes="*,Auto, Auto">
```
Comment (carldebilly) [inline src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleChooserControl.xaml]:
```suggestion
									  Sizes="*,Auto,Auto">
```

File: src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleControlInfoAttribute.cs [hunk 3]
```diff
@@ -6,36 +6,60 @@ namespace Uno.UI.Samples.Controls
 {
 
 	[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = true)]
```
Comment (carldebilly) [inline src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleControlInfoAttribute.cs]: `AllowMultiple = true` ?

Comment (dr1rrb) [reply src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Controls/SampleControlInfoAttribute.cs]: Agree it's a non sense, that's why I set it to false on the new attribute. But I won't change the old one.

File: doc/articles/uno-development/working-with-the-samples-apps.md (hunk 1 above)
Comment (dr1rrb) [reply doc/articles/uno-development/working-with-the-samples-apps.md]: It's compliant to point 6:
> A semantic line break MAY occur after a dependent clause in order to clarify grammatical structure or satisfy line length constraints.

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 4]
```diff
@@ -54,6 +54,7 @@
 - Enable partial `NavigationView.ItemSource` scenario (https://github.com/unoplatform/uno/issues/2477)
 - [Wasm] Fail gracefully if IDBFS is not enabled in emscripten
 - [#2513] Fix `TransformGroup` not working
+- [#1956] Fis iOS invalid final state when switching visual state before current state's animation is completed.
```
Comment (jeromelaban) [inline doc/ReleaseNotes/_ReleaseNotes.md]:
```suggestion
- [#1956] Fix iOS invalid final state when switching visual states before current state's animation is completed.
```

File: doc/articles/uno-development/working-with-the-samples-apps.md [hunk 5]
```diff
@@ -3,11 +3,14 @@
 The Uno solution provides a set of sample applications that provide a way to test features, as
 well as provide a way to write UI Tests.
 
-Those applications are structured in a way that samples can created out of normal `UserControl` instances, marked with the `SampleControlInfoAttribute` so the sample application can discover them.
+Those applications are structured in a way that samples can created out of normal `UserControl` instances,
```
Comment (jeromelaban) [inline doc/articles/uno-development/working-with-the-samples-apps.md]:
```suggestion
Those applications are structured in a way that samples can be created out of normal `UserControl` instances, 
```

File: doc/articles/uno-development/working-with-the-samples-apps.md [hunk 6]
```diff
@@ -20,7 +23,11 @@ The goal of these UI Tests is to :
 To create a UI Test for the sample applications:
 - Create or reuse a folder named from the namespace of the control or class your want to test, replacing "`.`" by "`_`"
 - Create a new `UserControl` from the Visual Studio templates in the `UITests.Shared` project
-- Add `[Uno.UI.Samples.Controls.SampleControlInfo("Replace_with_control_or_class_name", "MyTestName", description: "MyDescription")]` on the code-behind class.
+- Add `[Uno.UI.Samples.Controls.Sample]` on the code-behind class. 
+
+  _Note: This attribute also accepts few other metadata about you sample like a list of `Categories`, a `Name`,
```
Comment (jeromelaban) [inline doc/articles/uno-development/working-with-the-samples-apps.md]:
```suggestion
  _Note: This attribute also accepts a few other metadata about your sample, such as a list of `Categories`, a `Name`,
```

**Pull request merged**